platform/core/src/poly_out.rs

```diff
@@ -0,0 +1,112 @@
+use std::fs::DirBuilder;
+use std::path::Path;
+use uuid::Uuid;
+
+type ResourceId = String;
+type FsId = String;
+
+struct Metadata {}
+struct Content {}
+
+trait PolyOutTrait {
+    fn metadata(resource_id: ResourceId) -> Result<Metadata, String>;
+    fn read(resource_id: ResourceId) -> Result<Content, String>;
+    fn remove(resource_id: ResourceId) -> Result<(), String>;
+    // import will decide if it needs to unzip a file. The contents of the url will be extracted and inserted into a database.
+    fn import(
+        url: String,
+        feature_name: String,
+        dest_resource_id: Option<ResourceId>,
+    ) -> Result<ResourceId, String>;
+}
+
+struct PolyOut {}
```

Comment:
Like I said before, if we want to redesign this interface, we definitely shouldn't change it's name for the worse: It's currently called `FS` but exposed as `PolyOut`. I'd go for `FileSystem` I guess.

Reply:
OFS?

Reply:
Oracle File System? :D I think it's quite aligned with the `FileSystem` web API of the same name: https://developer.mozilla.org/en-US/docs/Web/API/FileSystem

Reply:
Object file system, rather. It gives you the chance to have a "typed" filesystem. But I'm good with Something-FileSystem, as long as we don't use the same name here and in the standard.

Reply:
FileSystem is the best option in my opinion. Let's end the confusion.